src/puppetlabs/puppetdb/admin.clj

```diff
@@ -0,0 +1,53 @@
+(ns puppetlabs.puppetdb.admin
+  (:require [clojure.java.io :as io]
+            [clojure.tools.logging :as log]
+            [compojure.core :as compojure]
+            [puppetlabs.kitchensink.core :as kitchensink]
+            [puppetlabs.puppetdb.archive :as archive]
+            [puppetlabs.puppetdb.cheshire :as json]
+            [puppetlabs.puppetdb.cli.export :as export]
+            [puppetlabs.puppetdb.cli.import :as import]
+            [puppetlabs.puppetdb.http :as http]
+            [puppetlabs.puppetdb.middleware :as mid]
+            [puppetlabs.trapperkeeper.core :as trapperkeeper]
+            [ring.middleware.multipart-params :as mp]
+            [clj-time.core :refer [now]]
+            [ring.util.io :as rio]))
+
+(def query-api-version :v4)
+
+(def munge-nodes-data
+  (comp (partial kitchensink/mapvals str)
+        #(select-keys % [:certname :facts_timestamp :catalog_timestamp :report_timestamp])))
+
+(defn export-app
+  [buffer query-fn]
+  (export/export! buffer
+                  (->> (query-fn :nodes query-api-version nil nil doall)
+                       (map munge-nodes-data)
+                       (mapcat (partial export/get-node-data query-fn)))))
+
+(defn build-app
+  [submit-command-fn query-fn {:keys [authorizer]}]
+  (-> (compojure/routes
+       (mp/wrap-multipart-params
+        (compojure/POST "/v1/archive" [archive command_versions]
+                        (do (import/import! (:tempfile archive)
```

Comment:
Event so, the user could throw an arbitrary file on the system at our import function, which isn't really what we want to do. I think this code could be a fair bit tighter on that front. 

One low-impact approach may be to provide our own handler function to wrap-multipart-params which wraps the default temp-file handler, tacking on some metadata, then checking that metadata inside the handler to verify that the data actually came from the place we expect.